Lib/test/test_httpservers.py

```diff
@@ -1535,6 +1537,197 @@ def test_server_test_ipv4(self, _):
+                                      tls_password_option,
+                                      self.tls_password_file)
+                    self.args['tls_cert'] = self.tls_cert
+                    self.args['tls_key'] = self.tls_key
+                    self.args['tls_password'] = self.tls_password
+                    mock_func.assert_called_once_with(**self.args)
+                    self.args['tls_cert'] = None
+                    self.args['tls_key'] = None
+                    self.args['tls_password'] = None
+                    mock_func.reset_mock()
+
+        # Abnormal: --tls-key without --tls-cert
+
+        for tls_key_option in tls_key_options:
+            for tls_cert_option in tls_cert_options:
+                with self.assertRaises(SystemExit):
+                    self.invoke_httpd(tls_key_option, self.tls_key)
+                    mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file without --tls-cert
+
+        for tls_password_option in tls_password_options:
+            with self.assertRaises(SystemExit):
+                self.invoke_httpd(tls_password_option, self.tls_password_file)
+                mock_func.reset_mock()
+
+        # Abnormal: --tls-password-file cannot be opened
+
+        non_existent_file = self.get_random_temporary_file_name()
+        retry_count = 0
+        while os.path.exists(non_existent_file) and retry_count < 10:
+            non_existent_file = self.get_random_temporary_file_name()
+        if not os.path.exists(non_existent_file):
+            for tls_password_option in tls_password_options:
+                for tls_cert_option in tls_cert_options:
+                    with self.assertRaises(SystemExit):
+                        self.invoke_httpd(tls_cert_option,
+                                          self.tls_cert,
+                                          tls_password_option,
+                                          non_existent_file)
+
+    def test_no_arguments(self, mock_func):
+        self.invoke_httpd()
+        mock_func.assert_called_once_with(HandlerClass=self.default_handler,
+                                          ServerClass=self.default_server,
+                                          protocol=self.default_protocol,
+                                          port=self.default_port,
+                                          bind=self.default_bind,
+                                          tls_cert=None,
+                                          tls_key=None,
+                                          tls_password=None)
+        mock_func.reset_mock()
+
+    def test_help_flag(self, _):
+        options = ['-h', '--help']
+        for option in options:
+            with self.assertRaises(SystemExit):
+                output = self.invoke_httpd(option)
+                self.assertStartsWith(output, 'usage: ')
+
+    def test_unknown_flag(self, _):
+        with self.assertRaises(SystemExit):
+            self.invoke_httpd('--unknown-flag')
```

Comment:
we should get help message, we can check it:
```suggestion
            output = self.invoke_httpd('--unknown-flag')
            self.assertStartsWith(output, 'usage: ')
```